pkg/util/errors/errors.go

```diff
@@ -5,13 +5,20 @@ import (
 	"net/url"
 	"os"
 	"syscall"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
 )
 
 // IsMatchingSyscallError returns true when the error is one of the Errno's in match
 // it deals with the different ways of wrapping the syscalls.
 func IsMatchingSyscallError(err error, match ...syscall.Errno) bool {
 	for {
 		switch t := err.(type) {
+		case *azure.RequestError:
+			err = &t.DetailedError
```

Comment:
@asalkeld shouldn't this be `t.DetailedError.Original`?

Reply:
it's caught in the line below
```
		case *autorest.DetailedError:
			err = t.Original
```
so that you can unwrap as many error combinations as possible.